### PR TITLE
JSON log format registration for kubelet

### PR DESCRIPTION
**Pull request by mengjiao-liu**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Currently importing k8s.io/component-base/logs results both text and json logging format available by default. As Json format is not yet GAed, it is undesired to force it on component authors that use the library. Instead it should be optional. 

As this change requires high level approvals in cmd directories it's recommended to do the change in multiple PRs，the PR only modifies the `kubelet` JSON log format registration.
- Add import _ "k8s.io/component-base/logs/json/register" to system components
  + kubelet

xref   #102644
#### Which issue(s) this PR fixes:

Part of #102632

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/apis/config/validation/validation.go [hunk 1]
```diff
@@ -25,6 +25,8 @@ import (
 	utilvalidation "k8s.io/apimachinery/pkg/util/validation"
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
 	"k8s.io/component-base/logs"
+	// As the JSON format is not yet GAed, an explicit registration is required
```
Comment (serathius) [inline pkg/kubelet/apis/config/validation/validation.go:28]: Please put the line in ./cmd/kubelet/kubelet.go

Comment (mengjiao-liu) [reply pkg/kubelet/apis/config/validation/validation.go]: Updated. Thanks! 😄

**Comment by thockin** [comment]
/approve

@serathius holds the LGTM just in case I am missing some detail here.

**Pull request merged**